src/osd/osd_op_util.cc

```diff
@@ -91,6 +103,9 @@ int OpInfo::set_from_op(
   const OSDMap &osdmap)
 {
   vector<OSDOp>::const_iterator iter;
+  int read_op_count = 0;
+  int stat_op_count = 0;
+  bool may_read_data = false;
```

Comment:
Not following the need for any of these three variables.  Can't `set_may_read_data` method just be called directly when a non-stat read op is encountered?

Reply:
Yes, I think you're right now.
It took me some time to remember why I did it in this way last year. I meant to limit the code change to, and only to, pure stat+write op and shouldn't influence others.
But as long as we name it as may_read_data(), I think it can be removed. See my last commit.